source/common/access_log/grpc_access_log_impl.cc

```diff
@@ -120,6 +120,12 @@ void HttpGrpcAccessLog::responseFlagsToAccessLogResponseFlags(
   if (request_info.getResponseFlag(RequestInfo::ResponseFlag::RateLimited)) {
     common_access_log.mutable_response_flags()->set_rate_limited(true);
   }
+
+  if (request_info.getResponseFlag(RequestInfo::ResponseFlag::UnauthorizedExternalService)) {
+    common_access_log.mutable_response_flags()->mutable_unauthorized_details()->set_reason(
```

Comment:
It's a bit weird that we have "details" in a "response_flags" message, but that's an API level concern. I wonder if we should rename that before moving it out of "not implemented" status. Thoughts?

Reply:
may be rename "_details" to "_reasons". That way we can set multiple reasons under it?